### ISSUE TITLE
SigMF Needs a Logo

### ISSUE THREAD

**Issue by bhilburn** [gnuradio/SigMF#117]
Pretty straight-forward, but the project needs a decent logo. I've got a few ideas, but would also happily entertain other ideas, especially from folks more artistic than I am (which frankly is a low bar 😅).

If you know a thing or two about graphic design and are up for taking this on, please post and let me know!

**Comment by Teque5** [comment]
3 min in inkscape
![2020-10_SigMF_v0](https://user-images.githubusercontent.com/784749/95289647-26ed6580-0820-11eb-9b4a-d3bae96eceb4.png)
an hour in inkscape
![2020-10_SigMF_v2_render](https://user-images.githubusercontent.com/784749/95289594-fc031180-081f-11eb-9f43-30b4681b4dfa.png)

**Comment by gmabey** [comment]
I vote for the 2nd (yellow/black) one.

**Comment by 777arc** [comment]
So the girl who created the CGRAN logo did it for free as long as we credited her, and she's actually been very pleased with the enormous number of places it ended up getting used, like on banners and websites other than CGRAN, so I bet she would be happy to try something out for SigMF.  We could either let her do whatever she feels like, or give her some inspiration like we did for CGRAN.  Let me know if we want to explore that option.

**Comment by jacobagilbert** [comment]
@777arc I think we should ask her.

**Comment by Teque5** [comment]
Ok I threw away a bit of time in blender:
![0001](https://user-images.githubusercontent.com/784749/119771329-61207b80-be72-11eb-81c6-93aff585ab12.png)

**Comment by 777arc** [comment]
Ooo that's good

**Comment by Teque5** [comment]
Okay I was thinking about this today and came up with a great idea. The logo will actually be in a `sigmf-data` & `sigmf-meta` file pair. I encoded the following two options (A & B) into a oscillioscope XY stereo audio signal. 
![image](https://user-images.githubusercontent.com/784749/119920394-23c6f700-bf21-11eb-89d5-156180057156.png)
A font is [justabit](http://www.pentacom.jp/pentacom/bitfontmaker2/gallery/?id=179). B is Eurostile. I was originally going to oscillioscope this 3d model, but it doesn't sound that great: 
![image](https://user-images.githubusercontent.com/784749/119920865-ddbe6300-bf21-11eb-95ae-d90deb06cb2f.png)

**Comment by jacobagilbert** [comment]
Lol I love this ☝️. Yes!

**Comment by Teque5** [comment]
Okay for something like this I can create a metadata file that has 2 annotations: one for the logo and one for the short shuttle at the end. Ideally it would serve as an example too.

https://user-images.githubusercontent.com/784749/119933429-eff7cb80-bf38-11eb-8938-6856a478ea98.mp4

**Comment by Teque5** [comment]
Here is a still from the Vectorscope:
![image](https://user-images.githubusercontent.com/784749/119933817-a9ef3780-bf39-11eb-8f4e-9272325a0d23.png)
And here is just a compact SVG render:

![00033a](https://user-images.githubusercontent.com/784749/119935012-a197fc00-bf3b-11eb-96f7-2ac4442809fa.png)

**Comment by bhilburn** [comment]
![](https://media.giphy.com/media/oYtVHSxngR3lC/giphy.gif)

holy CRAP @Teque5 this is amazing. I mean, just, wow. This is obviously the new logo 😄

How large are the associated data files (not the SVG caps)? Thinking through the best way to host them. It might be that the Wiki for this project is the best place to embed. And did you use a tool for creating the MP4?

I kind of want to use the vector scope still as the "official" logo, because its awesome. I wonder how hard it would be to turn that into an SVG. Not withstanding that, the "clean" SVG you made would also work great.

@Teque5 - This is seriously amazing 🤩

**Comment by jacobagilbert** [comment]
Personally, I like the 48kHz version.

These are so damn cool.

**Comment by bhilburn** [comment]
@Teque5 - Putting thoughts into bullets for ease of reading 🙂

* I agree with @jacobagilbert. I'm partial to the 48 kHz version... I kind of like that the Gibbs effects are a little more pronounced while still maintaining some of the "cleanliness".
* And great news re: file sizing. I don't want to put the SigMF recording into the repository, but with a filesize that low, we can easily attach it both (a) here in this issue and (b) as part of the file release bundle. I **would** like to put the SVG version of the logo into the repository.

Can you create:
* New SigMF recording without the shuttle
* MP4 video of that recording (looks like it would be ~6s long or so?)
* SVG of the still frame version of the logo

And upload them here? 🙂 Obviously, please feel free to share any other thoughts / ideas, too.

**Comment by Teque5** [comment]
In the attached preview [sigmf-logo.zip](https://github.com/gnuradio/SigMF/files/6602176/sigmf-logo.zip) are:

- (1) annotated `sigmf` pair
- (4) versions of the logo in `SVG`
- (1) `png` icon for discord servers
- (4) color videos of the 192KHz `wav` rendered in vectorscope
- (1) color video of the 48KHz `wav` rendered in oscillioscope
- (1) manifold `stl` with the logo for 3d printing
- (1) 48KHz `wav` recording
- (1) 192KHz `wav` recording

This is almost final but I have a few questions/issues:

- What license should I put in the sigmf file? Most of GR is GPL, but I think [cc by-sa 4.0](https://creativecommons.org/licenses/by-sa/4.0/) fits here?
- I encoded the left channel as 'real' and the right as 'imag' into a ci16 file. I wonder if this would confuse a newb. Is there any better way to write a stereo pair to `sigmf`?
- Any other annotations / fields we want to add?
- Someone else (@doug-aero, @ke8ctn}, @ericjmcd, @markubiak) should triple check my sigmf file pair.

In the final release version I'll include the a 192 KHz sigmf file too.

For those of you who just want to see the video, here it is:

https://user-images.githubusercontent.com/784749/120884041-92e1c280-c595-11eb-8880-0109e07fb071.mp4

_remember unmute_

**Comment by gmabey** [comment]
@Teque5 I think it's great -- I just wonder what will need to be done in order to have a recognizable thumbnail-sized logo, like if someone wanted to associate an icon with `.sigmf` file types ...

**Comment by Teque5** [comment]
@gmabey At very low resolutions the included `logo-filled.svg` looks good:
![icon](https://user-images.githubusercontent.com/784749/121095707-aebbb300-c7a5-11eb-8ed5-ae2f03dd026d.png)
For me the icon should actually the be spectrogram instead. _future project_

**Comment by bhilburn** [comment]
@Teque5 - This is fantastic. I'm honestly not sure how many times I've watched all of these videos at this point, but it's a lot.

Answering questions:

>     What license should I put in the sigmf file? Most of GR is GPL, but I think cc by-sa 4.0 fits here?
>     I encoded the left channel as 'real' and the right as 'imag' into a ci16 file. I wonder if this would confuse a newb. Is there any better way to write a stereo pair to sigmf?
>     Any other annotations / fields we want to add?
>     Someone else (@doug-aero, @ke8ctn}, @ericjmcd, @markubiak) should triple check my sigmf file pair.

* Agreed, CC-BY-SA 4.0 is a good fit, here.
* I totally agree, that doesn't make a ton of sense, would probably confuse a noob, and honestly, this feels to me like a bug (maybe feature enhancement, really) for SigMF that we ought to address. @Teque5, I'll open a new Issue to discuss this.

**Comment by Teque5** [comment]
In the attached final(?) [20210618_sigmf-release.zip](https://github.com/gnuradio/SigMF/files/6679721/20210618_sigmf-release.zip) are:

- (1) annotated `sigmf` pair **sigmf_logo.sigmf** containing master 48 KHz logo
- (1) annotated `sigmf` pair **sigmf_logo_fullband.sigmf** containing bonus 192 KHz logo
- (6) versions of the logo in `SVG`
- (7) `png`s for discord and folder icons
- (4) color videos of the 192KHz `wav` rendered in vectorscope
- (1) color video of the 48KHz `wav` rendered in oscillioscope
- (1) manifold `stl` with the logo for 3d printing
- (2) `wav` recordings at both rates

The main **new** thing added here is the *scope* version of the `svg`/`png` that I painstakingly rendered in python and saved as an `svg`. It's quite large (almost 1MB) due to the complexity but looks good. Here is a raster:
![logo-scope-256-A](https://user-images.githubusercontent.com/784749/122625774-ac860e00-d05b-11eb-9fce-fd7885a53d09.png)

The scope `svg` was created using the code in [this gist](https://gist.github.com/Teque5/3a377440916506fb2ac46a9f47251da4).

I also improved the discord icon (and tested w/light & dark mode) so it looks nice too:
![logo-icon-32-discord](https://user-images.githubusercontent.com/784749/122625809-e22af700-d05b-11eb-813d-5ffa6faadd4b.png)

I think this is final unless we want to alter the `sigmffile` for multichannel recordings per #133 in the future.

**Comment by Teque5** [comment]
If the logo is committed to the repo, I propose we remove the `example_metadata.py` from the top level since it's outdated anyway.

**Comment by bhilburn** [comment]
@Teque5 - I don't think I could be more excited about this. It's just so. freaking. good. Also, `logo-scope-256-B` is my favorite static image - well done 👏👏👏

I want to get #138 merged to close #133. Can you do the following:

* Update the metadata files in this zip to reflect the new `num_channels` field from #138, upload the new zip here.
* File a PR that updates the top-level README.md to put `logo-scope-256-B at the top of the page, and deletes `example_metadata.py` and replaces it with the metadata file for the logo.

I will get the zip archive into the repo's wiki for permanent storage, update site resources, and then update the README.md (following your PR) to add additional detail.

This is awesome, @Teque5. You /know/ one of these videos is making a cameo at GRCon'21 😁

**Comment by bhilburn** [comment]
(By the way, feel free not to wait for #138 to get merged to do the above. That part of the PR isn't going to change.)

**Comment by bhilburn** [comment]
Something else occurred to me: we should set the "core:version" field to `1.0.0`. Doesn't technically exist, yet, but it will, and the num_channels field doesn't make sense without it.

**Comment by Teque5** [comment]
Okay I'll create the PR. Here is the newest [20210702_release.zip](https://github.com/gnuradio/SigMF/files/6757667/20210702_release.zip) with your requested changes. I included an `svg` of the color logo you liked along with slightly improved versions of a few files.
![logo-scope-color](https://user-images.githubusercontent.com/784749/124339392-14138180-db63-11eb-9dc4-e4ff678e0f97.png)

**Comment by bhilburn** [comment]
It's perfect! 😍

Okay, I got the zip file uploaded into the SigMF project's wiki's git repo, and linked to it here:
https://github.com/gnuradio/SigMF/wiki/SigMF-Logo

@Teque5 - Have a look at that page, and feel free to edit it as you'd like! I expect the logo will generate a ton of interest, and I think the more detail we have there, the better 🙂

Closing this issue. I'm adding the logo to the front-page README, now, and merging #143 will be the last step in this!
